traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/QuestionsActivity.java

```diff
@@ -85,9 +91,26 @@ protected void onSaveInstanceState(@NonNull Bundle outState) {
 
     @Override
     public void onOrderCreateSuccess(Order order) {
-        Intent intent = new Intent(this, OrderSummaryActivity.class);
-        intent.putExtra(OrderSummaryActivity.EXTRA_ORDER, order);
-        startActivityForResult(intent, REQUEST_CODE_ORDER_FLOW);
+        TravelerUI.getPaymentManager().fetchPayments(new PaymentsFetchCallback() {
+            @Override
+            public void onPaymentsFetchSuccess(List<Payment> paymentList) {
+                advanceToSummary(new ArrayList<>(paymentList));
+            }
+
+            @Override
+            public void onPaymentsFetchError(Error error) {
+                // Graceful failure: In case we can't fetch saved payments we still
+                // want to advance the user
+                advanceToSummary(new ArrayList<>());
```

Comment:
Might make sense to make `payments` Nullable so `OrderSummaryActivity` can display an appropriate error message